src/ExcelProvider.Runtime/ExcelProvider.Runtime.fs

```diff
@@ -373,7 +383,9 @@ type ExcelFileInternal private (view, documentId, sheetname, hasheaders) =
         let buildRow rowIndex =
             new Row(documentId, sheetname, rowIndex, getCellValue view, columns)
 
-        seq { (if hasheaders then 1 else 0) .. view.RowCount } |> Seq.map buildRow
+        let zeroBasedLastIndex = view.RowCount - 1
+
+        seq { (if hasheaders then 1 else 0) .. zeroBasedLastIndex } |> Seq.map buildRow
```

Comment:
And finally here, if `View.RowCount` suggested 3 rows, this would grab rows from indices 0,1,2,3, for four rows, as (a .. b) is min and max inclusive. 

And we're back to an `Unbounded` `Range` having one extra row, while a `Bounded` `Range` would have the correct number of rows.